packages/sit-onyx/src/utils/playwright.tsx

```diff
@@ -90,50 +86,70 @@ type CaseBuilder<S extends ComponentStates> = (
  *       if (focusState === "hover") await input.hover();
  *       return component;
  *     },
- *   ),
- * );
+ *   );
+ * });
  * ```
  *
  * @param states All possible states for which permutations will be generated. Use `as const` to allow type support for the values.
  * @param baseName Prefix of the generated screenshot file names.
  * @param caseBuilder Build function that will be called for every permutation to generate JSX and perform setup interactions for the given component state.
  */
-export const createScreenshotsForAllStates =
-  <S extends Readonly<Record<string, ReadonlyArray<string>>>>(
-    states: S,
-    baseName: string,
-    caseBuilder: CaseBuilder<S>,
-  ) =>
-  async ({ mount, page }: TestArg) => {
-    const permutations = generatePermutations(states);
+export const executeScreenshotsForAllStates = <
+  S extends Readonly<Record<string, ReadonlyArray<string>>>,
+>(
+  states: S,
+  baseName: string,
+  caseBuilder: CaseBuilder<S>,
+) => {
+  const permutations = generatePermutations(states);
 
-    for (const testCase of permutations) {
-      // ARRANGE
-      const wrappedMount = ((jsx, options) =>
-        mount(
-          <div
-            style={{ width: "max-content", padding: "1rem" }}
-            class={{
-              "onyx-use-optional": options?.useOptional,
-            }}
-          >
-            {jsx}
-          </div>,
-        )) satisfies WrappedMount;
+  // to prevent JavaScript memory heap errors we should not run all permutations inside
+  // one single Playwright test. On the other hand we should also not run a single
+  // test for every permutation because this leads to a very bad test performance.
+  // therefore, we will split the permutations to only run a maximum of X permutations inside a single test
+  const batches: (typeof permutations)[] = [];
+  while (permutations.length > 0) {
+    batches.push(permutations.splice(0, 100));
+  }
 
-      await page.getByRole("document").focus(); // reset focus
-      await page.getByRole("document").hover(); // reset mouse
-      await page.mouse.up(); // reset mouse
-      const component = await caseBuilder(testCase, wrappedMount, page);
+  batches.forEach((batch, index) => {
+    test(`${baseName} state screenshot tests (batch ${index + 1})`, async ({ mount, page }) => {
+      // default test timeout is 10s (see playwright.config.ts), so we increase it here
+      // so that the timeout is applied to every permutation
+      test.setTimeout(batch.length * 10 * 1000);
```

Comment:
10 seconds per permutation seems like a lot. That would be 15 mins for one batch. I would think 1 or 2 seconds should be more than enough for one permutation.

Reply:
Done